src/factories/locationFactory.ts

```diff
@@ -36,6 +39,39 @@ export namespace LocationFactory {
             uri = Utilities.combineUriWithUriBase(physLocation.artifactLocation.uri, uriBase);
 
             if (uri) {
+                // Check for artifact content and create our artifact content URI.
+                // In order to have artifact content, the log must have a valid artifact index
+                // (represented by artifact location index) and the "contents" property of the artifact
+                // must be defined and have one of the "binary, text, or rendered" properties set.
+                if (physLocation.artifactLocation.index !== undefined) {
+                    const run: sarif.Run | undefined = sarifLog.runs[runInfo.runIndex];
+                    if (run && run.artifacts) {
+                        const artifact: sarif.Artifact | undefined = run.artifacts[physLocation.artifactLocation.index];
+                        if (artifact) {
+                            const artifactContent: sarif.ArtifactContent | undefined = artifact.contents;
+                            if (artifactContent) {
+                                fileName = uri.toString(true).substring(uri.toString(true).lastIndexOf('/') + 1);
+
+                                // See if we have a custom renderer for an artifact content object (for example, binary content).
+                                const artifactContentRenderer: ArtifactContentRenderer | undefined = tryCreateRendererForArtifactContent(sarifLog, artifactContent, runInfo.runIndex, physLocation.artifactLocation.index);
+                                const artifactContentUri: Uri | undefined = ArtifactContentFileSystemProvider.tryCreateUri(sarifLog, Uri.file(runInfo.sarifFileFullPath), uri, runInfo.runIndex, physLocation.artifactLocation.index, artifactContentRenderer);
```

Comment:
>artifactContentRenderer [](start = 230, length = 23)

I want to change this parameter. The only coupling for URI creation is whether a specific extension is required for the URI (which in the case of binary rendering it needs to be ".md")